src/SDKs/Batch/DataPlane/Azure.Batch.Unit.Tests/BatchSharedKeyCredentialUnitTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Azure.Batch.Unit.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Batch.Protocol;
+    using Microsoft.Net.Http.Server;
+    using Xunit;
+
+    public class BatchSharedKeyCredentialUnitTests
+    {
+        private const string url = "http://localhost:2055";
+
+        [Theory]
+        [MemberData(nameof(HttpMethods))]
+        public async Task TestSharedKeySigning_ContentLength(HttpMethod httpMethod)
+        {
+            BatchSharedKeyCredential creds = new BatchSharedKeyCredential(ClientUnitTestCommon.DummyAccountName, ClientUnitTestCommon.DummyAccountKey);
+
+            HttpRequestMessage message = new HttpRequestMessage(httpMethod, url);
+            message.Headers.Add("client-request-id", Guid.NewGuid().ToString());
+
+            await creds.ProcessHttpRequestAsync(message, CancellationToken.None);
+            Assert.NotNull(message.Headers.Authorization);
+
+            void AssertLambda(RequestContext ctx)
+            {
+                if (httpMethod == HttpMethod.Head || httpMethod == HttpMethod.Get)
```

Comment:
While these tests are good, they test an aspect of the underlying behavior, but don't seem to actually test our shared key algorithm. What we're wanting to do is enforce that the StringToSign that we computed (made up of headers etc) matches the StringToSign that would be computed by the server (based on the headers sent).
So we'd either need a way of extracting the StringToSign from our BatchSharedKeyCredential (do we expose this?) or else compute the hash again on the server side and validate that they match.

I think we should keep these tests too as a way of exploring the underlying functionality / confirming that different platforms don't change behavior, however the "SharedKeyAuth" doesn't seem to belong inside this test as it isn't being asserted against.